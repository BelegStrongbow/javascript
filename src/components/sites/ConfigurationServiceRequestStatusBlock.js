import PropTypes from "prop-types";
import React from "react";
import { defineMessages, FormattedMessage, FormattedHTMLMessage, injectIntl, intlShape } from "react-intl";
import { Paper, WhitePage } from "../PaperStyles";
import CollapsibleHeader from "../CollapsibleHeader";

let messages = defineMessages( {
	configurationPendingHeading: {
		id: "requestConfigurationStatus.configurationPendingHeading",
		defaultMessage: "Configuration service successfully requested",
	},
	configurationPendingMessage: {
		id: "requestConfigurationStatus.configurationPendingMessage",
		defaultMessage: "Thank you for submitting the intake form, we'll take it from here and complete the SEO" +
		" configuration for this website within <strong>three business days</strong>. If you have any questions" +
		" about this process, take a look at our <a href='{ knowledgeBaseLink }'>Knowledge Base</a>.",
	},
	configurationCompletedHeading: {
		id: "requestConfigurationStatus.configurationCompletedHeading",
		defaultMessage: "Configuration service completed",
	},
	configurationCompletedMessage: {
		id: "requestConfigurationStatus.configurationCompletedMessage",
		defaultMessage: "Your configuration service has been successfully completed!",
	},
	configurationStatus: {
		id: "requestConfigurationStatus.status",
		defaultMessage: "Current status: ",
	},
	configurationPending: {
		id: "requestConfigurationStatus.pending",
		defaultMessage: "pending",
	},
	configurationInProgress: {
		id: "requestConfigurationStatus.inProgress",
		defaultMessage: "in progress",
	},
	configurationCompleted: {
		id: "requestConfigurationStatus.completed",
		defaultMessage: "completed",
	},
} );

const STATUSES = {
	submitted: "configurationPending",
	"in progress": "configurationInProgress",
	completed: "configurationCompleted",
};

class ConfigurationServiceRequestStatusBlock extends React.Component {
	getCurrentStatus() {
		if ( this.props.status === "completed" ) {
			return null;
		}

		return (
			<p>
				<FormattedMessage
					id={ messages.configurationStatus.id }
					defaultMessage={ messages.configurationStatus.defaultMessage }
				/>
				{
					Object.keys( STATUSES ).map( ( status ) => {
						let formattedMessage = (
							<FormattedMessage
								key={ status }
								id={ messages[ STATUSES[ status ] ].id }
								defaultMessage={ messages[ STATUSES[ status ] ].defaultMessage }
							/>
						);

						if ( status === this.props.status ) {
							return <strong key={ status }>{ formattedMessage }</strong>;
						}

						return formattedMessage;
					} ).reduce( ( prev, curr, i ) => [ prev, <span key={ i }> / </span>, curr ] )
				}
			</p>
		);
	}

	render() {
		let headerMessage  = messages.configurationPendingHeading;
		let contentMessage = messages.configurationPendingMessage;
		if ( this.props.status === "completed" ) {
			headerMessage  = messages.configurationCompletedHeading;
			contentMessage = messages.configurationCompletedMessage;
		}

		return (
			<Paper>
				<CollapsibleHeader
					title={ this.props.intl.formatMessage( headerMessage ) }
					isOpen={ this.props.status !== "completed" }
				>
					<WhitePage>
						<div>
							<p>
								<FormattedHTMLMessage
									id={ contentMessage.id }
									defaultMessage={ contentMessage.defaultMessage }
									values={ {
										knowledgeBaseLink: "https://kb.yoast.com",
									} }
								/>
							</p>
							{ this.getCurrentStatus() }
						</div>
					</WhitePage>
				</CollapsibleHeader>
			</Paper>
		);
	}
}

ConfigurationServiceRequestStatusBlock.propTypes = {
	status: PropTypes.string.isRequired,
	intl: intlShape.isRequired,
};

export default injectIntl( ConfigurationServiceRequestStatusBlock );
