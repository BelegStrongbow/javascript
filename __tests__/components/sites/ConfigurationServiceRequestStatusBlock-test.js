import React from 'react';
import ConfigurationServiceRequestStatusBlock from '../../../src/components/sites/ConfigurationServiceRequestStatusBlock';
import { createComponentWithIntl } from "../../../utils";

test('the configuration service request status block component with status submitted matches the snapshot', () => {
	const component = createComponentWithIntl(
		<ConfigurationServiceRequestStatusBlock id="test" status="submitted" />
	);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('the configuration service request status block component with status in progress matches the snapshot', () => {
	const component = createComponentWithIntl(
		<ConfigurationServiceRequestStatusBlock id="test" status="in progress" />
	);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('the configuration service request status block component with status completed matches the snapshot', () => {
	const component = createComponentWithIntl(
		<ConfigurationServiceRequestStatusBlock id="test" status="completed" />
	);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
