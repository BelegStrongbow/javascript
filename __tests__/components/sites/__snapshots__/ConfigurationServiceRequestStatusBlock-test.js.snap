// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the configuration service request status block component with status completed matches the snapshot 1`] = `
.c0 {
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.2);
}

.c1 {
  margin-top: 24px;
  background-color: #fff;
  width: 100%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fff;
  padding: 24px 32px;
  border: none;
  text-align: left;
  font-weight: 300;
  cursor: pointer;
  color: #000;
}

.c2 svg {
  -webkit-flex: 0 0 40px;
  -ms-flex: 0 0 40px;
  flex: 0 0 40px;
  margin-right: -10px;
  margin-left: 10px;
  margin-top: -4px;
  margin-bottom: -4px;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
}

.c3 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  font-size: 1.5em;
  line-height: 1.33333333;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

@media screen and ( max-width:1355px ) {
  .c2 {
    padding: 16px 24px;
  }
}

@media screen and ( max-width:800px ) {
  .c2 {
    padding: 16px;
  }
}

@media screen and ( max-width:1355px ) {
  .c2 svg {
    margin-top: 4px;
    margin-bottom: 4px;
    margin-right: -2px;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <button
      aria-expanded={false}
      className="c2"
      fontWeight={300}
      onClick={[Function]}
    >
      <span
        className="c3"
        fontSize={1.5}
      >
        Configuration service completed
      </span>
      <svg
        aria-hidden="true"
        fill="#555"
        focusable="false"
        height={40}
        role="img"
        viewBox="0 0 24 24"
        width={40}
      >
        <path
          d="M7.41 15.41L12 10.83l4.59 4.58L18 14l-6-6-6 6z"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`the configuration service request status block component with status in progress matches the snapshot 1`] = `
.c0 {
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.2);
}

.c4 {
  background-color: #fff;
  padding: 16px 24px;
  margin-top: 24px;
}

.c1 {
  margin-top: 24px;
  background-color: #fff;
  width: 100%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fff;
  padding: 24px 32px;
  border: none;
  text-align: left;
  font-weight: 300;
  cursor: pointer;
  color: #000;
}

.c2 svg {
  -webkit-flex: 0 0 40px;
  -ms-flex: 0 0 40px;
  flex: 0 0 40px;
  margin-right: -10px;
  margin-left: 10px;
  margin-top: -4px;
  margin-bottom: -4px;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
}

.c3 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  font-size: 1.5em;
  line-height: 1.33333333;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c6 {
  display: inline-block;
  height: 48px;
  padding: 12px 16px;
  background-color: #64a60a;
  color: #fff;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.2);
  border-radius: 4px;
  vertical-align: top;
  font: 700 14px/24px "Open Sans",sans-serif;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  text-shadow: 0px 0px 2px #000;
  border-bottom: 4px solid rgba(0,0,0,0.2);
  -webkit-transition: background-color 150ms ease-out;
  transition: background-color 150ms ease-out;
}

.c6:hover,
.c6:focus {
  color: #fff;
}

.c6:hover,
.c6:focus {
  box-shadow: 0 4px 10px 0 rgba(0,0,0,0.2),inset 0 0 0 100px rgba( 0,0,0,0.1 );
}

.c6:active {
  -webkit-transform: translateY( 1px );
  -ms-transform: translateY( 1px );
  transform: translateY( 1px );
  box-shadow: none;
}

.c5 {
  min-width: 152px;
}

@media screen and ( max-width:1355px ) {
  .c2 {
    padding: 16px 24px;
  }
}

@media screen and ( max-width:800px ) {
  .c2 {
    padding: 16px;
  }
}

@media screen and ( max-width:1355px ) {
  .c2 svg {
    margin-top: 4px;
    margin-bottom: 4px;
    margin-right: -2px;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <button
      aria-expanded={true}
      className="c2"
      fontWeight={300}
      onClick={[Function]}
    >
      <span
        className="c3"
        fontSize={1.5}
      >
        Configuration service successfully requested
      </span>
      <svg
        aria-hidden="true"
        fill="#555"
        focusable="false"
        height={40}
        role="img"
        viewBox="0 0 24 24"
        width={40}
      >
        <path
          d="M7.41 15.41L12 10.83l4.59 4.58L18 14l-6-6-6 6z"
        />
      </svg>
    </button>
    <div
      className="c4"
    >
      <div>
        <p>
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "Thank you for submitting the intake form, we'll take it from here and complete the SEO configuration for this website within <strong>three business days</strong>. If you have any questions about this process, take a look at our <a href='https://kb.yoast.com'>Knowledge Base</a>.",
              }
            }
          />
        </p>
        <a
          aria-label=""
          className="c5 c6"
          href="https://my.yoast.com/api/ConfigurationServiceRequests/test/report?access_token="
          onClick={null}
          rel="noopener"
          target="_blank"
        >
          View Report
        </a>
        <p>
          <span>
            Current status: 
          </span>
          <span>
            pending
          </span>
          <span>
             / 
          </span>
          <strong>
            <span>
              in progress
            </span>
          </strong>
          <span>
             / 
          </span>
          <span>
            completed
          </span>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`the configuration service request status block component with status submitted matches the snapshot 1`] = `
.c0 {
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.2);
}

.c4 {
  background-color: #fff;
  padding: 16px 24px;
  margin-top: 24px;
}

.c1 {
  margin-top: 24px;
  background-color: #fff;
  width: 100%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fff;
  padding: 24px 32px;
  border: none;
  text-align: left;
  font-weight: 300;
  cursor: pointer;
  color: #000;
}

.c2 svg {
  -webkit-flex: 0 0 40px;
  -ms-flex: 0 0 40px;
  flex: 0 0 40px;
  margin-right: -10px;
  margin-left: 10px;
  margin-top: -4px;
  margin-bottom: -4px;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
}

.c3 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  font-size: 1.5em;
  line-height: 1.33333333;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c6 {
  display: inline-block;
  height: 48px;
  padding: 12px 16px;
  background-color: #64a60a;
  color: #fff;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.2);
  border-radius: 4px;
  vertical-align: top;
  font: 700 14px/24px "Open Sans",sans-serif;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  text-shadow: 0px 0px 2px #000;
  border-bottom: 4px solid rgba(0,0,0,0.2);
  -webkit-transition: background-color 150ms ease-out;
  transition: background-color 150ms ease-out;
}

.c6:hover,
.c6:focus {
  color: #fff;
}

.c6:hover,
.c6:focus {
  box-shadow: 0 4px 10px 0 rgba(0,0,0,0.2),inset 0 0 0 100px rgba( 0,0,0,0.1 );
}

.c6:active {
  -webkit-transform: translateY( 1px );
  -ms-transform: translateY( 1px );
  transform: translateY( 1px );
  box-shadow: none;
}

.c5 {
  min-width: 152px;
}

@media screen and ( max-width:1355px ) {
  .c2 {
    padding: 16px 24px;
  }
}

@media screen and ( max-width:800px ) {
  .c2 {
    padding: 16px;
  }
}

@media screen and ( max-width:1355px ) {
  .c2 svg {
    margin-top: 4px;
    margin-bottom: 4px;
    margin-right: -2px;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <button
      aria-expanded={true}
      className="c2"
      fontWeight={300}
      onClick={[Function]}
    >
      <span
        className="c3"
        fontSize={1.5}
      >
        Configuration service successfully requested
      </span>
      <svg
        aria-hidden="true"
        fill="#555"
        focusable="false"
        height={40}
        role="img"
        viewBox="0 0 24 24"
        width={40}
      >
        <path
          d="M7.41 15.41L12 10.83l4.59 4.58L18 14l-6-6-6 6z"
        />
      </svg>
    </button>
    <div
      className="c4"
    >
      <div>
        <p>
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "Thank you for submitting the intake form, we'll take it from here and complete the SEO configuration for this website within <strong>three business days</strong>. If you have any questions about this process, take a look at our <a href='https://kb.yoast.com'>Knowledge Base</a>.",
              }
            }
          />
        </p>
        <a
          aria-label=""
          className="c5 c6"
          href="https://my.yoast.com/api/ConfigurationServiceRequests/test/report?access_token="
          onClick={null}
          rel="noopener"
          target="_blank"
        >
          View Report
        </a>
        <p>
          <span>
            Current status: 
          </span>
          <strong>
            <span>
              pending
            </span>
          </strong>
          <span>
             / 
          </span>
          <span>
            in progress
          </span>
          <span>
             / 
          </span>
          <span>
            completed
          </span>
        </p>
      </div>
    </div>
  </div>
</div>
`;
